include/sycl/algorithm/sort.hpp

```diff
@@ -173,6 +173,780 @@ void sequential_sort(cl::sycl::queue q, cl::sycl::buffer<T, 1, Alloc> buf,
+
+  using global_buffer_accessor_t =
+      cl::sycl::accessor<data_vec_type, 1, cl::sycl::access::mode::read_write,
+                         cl::sycl::access::target::global_buffer>;
+  using local_buffer_accessor_t =
+      cl::sycl::accessor<data_vec_type, 1, cl::sycl::access::mode::read_write,
+                         cl::sycl::access::target::local>;
+};
+
+template <typename T, unsigned U>
+class bitonic_sort_init : public bitonic_sort_base<T, U> {
+public:
+  bitonic_sort_init(
+      const typename bitonic_sort_base<T, U>::global_buffer_accessor_t
+          &globalBuf,
+      const typename bitonic_sort_base<T, U>::local_buffer_accessor_t &localBuf)
+      : m_globalBuf(globalBuf), m_localBuf(localBuf) {}
+
+  void operator()(cl::sycl::nd_item<1> item) {
+    using namespace cl::sycl;
+#ifdef EMULATE_SHUFFLE_BUILTINS
+    using namespace emulated_shuffle_builtins;
+#endif
+
+    typedef typename bitonic_sort_base<T, U>::data_vec_type data_vec_type_;
+    typedef
+        typename bitonic_sort_base<T, U>::mask_op_vec_type mask_op_vec_type_;
+    typedef typename bitonic_sort_base<T, U>::relational_op_vec_type
+        relational_op_vec_type_;
+
+    const int vec_elems_per_thread_ =
+        bitonic_sort_base<T, U>::vec_elems_per_thread_::value;
+
+    int dir;
+    unsigned int id, global_start, size, stride;
+
+    data_vec_type_ input1, input2, temp;
+
+    relational_op_vec_type_ comp;
+
+    mask_op_vec_type_ mask1(1, 0, 3, 2);
+    mask_op_vec_type_ mask2(2, 3, 0, 1);
+    mask_op_vec_type_ mask3(3, 2, 1, 0);
+
+    relational_op_vec_type_ add1(1, 1, 3, 3);
+    relational_op_vec_type_ add2(2, 3, 2, 3);
+    relational_op_vec_type_ add3(1, 2, 2, 3);
+
+    id = item.get_local_id(0) * vec_elems_per_thread_;
+    global_start =
+        item.get_group(0) * item.get_local_range(0) * vec_elems_per_thread_ +
+        id;
+
+    input1 = m_globalBuf[global_start];
+    input2 = m_globalBuf[global_start + 1];
+
+    /* Sort input 1 - ascending */
+    comp = (input1 < shuffle(input1, mask1));
+    input1 = shuffle(input1, (comp + add1).template as<mask_op_vec_type_>());
+    comp = (input1 < shuffle(input1, mask2));
+    input1 =
+        shuffle(input1, (comp * 2 + add2).template as<mask_op_vec_type_>());
+    comp = (input1 < shuffle(input1, mask3));
+    input1 = shuffle(input1, (comp + add3).template as<mask_op_vec_type_>());
+
+    /* Sort input 2 - descending */
+    comp = (input2 > shuffle(input2, mask1));
+    input2 = shuffle(input2, (comp + add1).template as<mask_op_vec_type_>());
+    comp = (input2 > shuffle(input2, mask2));
+    input2 =
+        shuffle(input2, (comp * 2 + add2).template as<mask_op_vec_type_>());
+    comp = (input2 > shuffle(input2, mask3));
+    input2 = shuffle(input2, (comp + add3).template as<mask_op_vec_type_>());
+
+    /* Swap corresponding elements of input 1 and 2 */
+    add3 = relational_op_vec_type_(4, 5, 6, 7);
+    dir = item.get_local_id(0) % vec_elems_per_thread_ * -1;
```

Comment:
Is there a motivation for the `*-1` compared to unary `-` in front and some parenthesis?